### DISCUSSION TITLE
Potential Sync Improvements for Separate Windows

### DISCUSSION THREAD

**Discussion by utkonos** [Vector35/binaryninja-api#5270] [Ideas]
When using multiple windows, it would be very helpful if the sync group could also apply to associated Stack and Variables panes. This probably should be an additional, optional configuration, not a change to default behavior. I can see contexts where it should be off as well.

Implementation Thoughts:

This could be handled by menu items in the pancake menu for a main window. Add a "Sync" menu item and then have on/off selections for the Stack and Variables pane located in the same window.

<img width="264" alt="image" src="https://github.com/Vector35/binaryninja-api/assets/2775739/b8ff79bd-a849-41ca-ae3e-43401e446058">

**Comment by fuzyll** [comment]
It's a little unclear to me what you want here.

If you have two windows and you *don't* want the Linear View in the second window to update when you change your location in the first window, you would need to change the sync groups of the two Linear Views to be separate.

If you have two windows and you *do* want the Linear View in the second window to update when you change your location in the first window, you would need to make sure that both Linear Views are in the same sync group.

You should be able to have both of these occur at the same time by splitting a Linear View and setting different sync groups for both. Then, you would get each behavior depending on where you had last navigated within that window.

Is there some other feature you want here that we are missing?

**Comment by utkonos** [comment]
Let me clarify: two separate windows one single sync group. In this state, I would like to click a location in window A and have the Stack and Variables panes for window B follow the location of focus if it is a variable. And if it is not a variable, not only the line of focus but also the item of focus follows as well.

"Window A" where I have selected a variable:
<img width="650" alt="image" src="https://github.com/Vector35/binaryninja-api/assets/2775739/b44f58f8-86ad-4db2-a012-1d36ef589962">

"Window B" where it would be helpful to follow the same selection of the variable (1) not just the line. The location 2 would also have selection focus put on it.
<img width="1339" alt="Screenshot 2024-04-16 at 16 48 22" src="https://github.com/Vector35/binaryninja-api/assets/2775739/0d6c0fd8-428e-4367-b7d0-93e166a6a319">

Then, also in "Window B", the Stack pane would also follow focus:
<img width="1271" alt="Screenshot 2024-04-16 at 16 48" src="https://github.com/Vector35/binaryninja-api/assets/2775739/ac1f809d-f2ef-4b52-b744-4e868df09c3f">

Does that explain it better?

**Comment by utkonos** [comment]
The only reason I would add a menu item is to retain the current behavior if someone needs it this way. Another option is just to modify how sync groups work as far as focus is concerned in the default and only way that it works now. Another option would be a single checkbox in Preferences that lets the user go back to the current sync group behavior if they wish (or vice versa).
